src/main/ruby/truffleruby/core/time.rb

```diff
@@ -43,14 +43,25 @@ class Time
   }
 
   def inspect
+    str = strftime('%Y-%m-%d %H:%M:%S')
+
+    if nsec != 0
+      str << sprintf(".%09d", nsec)
+      str.chop! while str.end_with?('0')
+    end
+
+    str << (gmt? ? ' UTC' : strftime(' %z'))
+    str.force_encoding Encoding::US_ASCII
+  end
+
+  def to_s
     if gmt?
       str = strftime('%Y-%m-%d %H:%M:%S UTC')
     else
       str = strftime('%Y-%m-%d %H:%M:%S %z')
     end
     str.force_encoding Encoding::US_ASCII
```

Comment:
It seems better to refactor `to_s` to also use the `str << (gmt? ? ' UTC' : strftime(' %z'))` approach to make comparison easier, I'll do that.

Reply:
Thank you. I wasn’t sure of the performance impact of mutating the string so I erred on the side of leaving it alone, but I agree it’s better to make it look the same if there’s no downside.